sdk/typescript/src/client/cryptography.ts

```diff
@@ -0,0 +1,358 @@
+		centralizedSecretKeyShare,
+		classGroupsKeypair.encryptionKey,
+		networkDecryptionKeyPublicOutput,
+	);
+
+	return {
+		centralizedPublicKeyShareAndProof: Uint8Array.from(centralizedPublicKeyShareAndProof),
+		centralizedPublicOutput: Uint8Array.from(centralizedPublicOutput),
+		centralizedSecretKeyShare: Uint8Array.from(centralizedSecretKeyShare),
+		encryptedUserShareAndProof: Uint8Array.from(encryptedUserShareAndProof),
+	};
+}
+
+/**
+ * Prepare verification data for importing an existing cryptographic key as a DWallet.
+ * This function creates all necessary proofs and encrypted data for the import process.
+ *
+ * @param ikaClient - The IkaClient instance to fetch network parameters from
+ * @param sessionIdentifier - Unique identifier for this import session
+ * @param userShareEncryptionKeys - The user's encryption keys for securing the imported share
+ * @param keypair - The existing Secp256k1 keypair to import as a DWallet
+ * @returns Promise resolving to complete verification data for the import process
+ * @throws {Error} If network parameters cannot be fetched or key import preparation fails
+ */
+export async function prepareImportDWalletVerification(
+	ikaClient: IkaClient,
+	sessionIdentifier: Uint8Array,
+	userShareEncryptionKeys: UserShareEncrytionKeys,
+	keypair: Secp256k1Keypair,
+): Promise<PreparedImportDWalletVerification> {
+	const networkDecryptionKeyPublicOutput = await ikaClient.getNetworkPublicParameters();
+
+	const [secret_share, public_output, outgoing_message] = create_imported_dwallet_centralized_step(
+		networkDecryptionKeyPublicOutput,
+		sessionIdentifierDigest(sessionIdentifier),
+		bcs.vector(bcs.u8()).serialize(decodeSuiPrivateKey(keypair.getSecretKey()).secretKey).toBytes(),
+	);
+
+	const encryptedUserShareAndProof = encryptSecretShare(
+		secret_share,
+		userShareEncryptionKeys.encryptionKey,
+		networkDecryptionKeyPublicOutput,
+	);
+
+	return {
+		secret_share: Uint8Array.from(secret_share),
+		public_output: Uint8Array.from(public_output),
+		outgoing_message: Uint8Array.from(outgoing_message),
+		encryptedUserShareAndProof: Uint8Array.from(encryptedUserShareAndProof),
+	};
+}
+
+/**
+ * Create the centralized output for a signature generation process.
+ * This function combines the user's secret key, presign, and message to create a signature component.
+ *
+ * @param networkDecryptionKeyPublicOutput - The network's public parameters
+ * @param activeDWallet - The active DWallet containing the public output
+ * @param secretKey - The user's secret key share
+ * @param presign - The presignature data from a completed presign operation
+ * @param message - The message bytes to sign
+ * @param hash - The hash scheme identifier to use for signing
+ * @returns The centralized signature output that can be combined with other signature components
+ * @throws {Error} If the DWallet is not in active state or public output is missing
+ */
+export function createSignCentralizedOutput(
```

Comment:
This doesn't create the output, it creates the user's sign message to be sent to the network. Rename please